Cargo.toml

```diff
@@ -1,6 +1,12 @@
 [workspace]
 resolver = "2"
-members = ["rs-matter", "rs-matter-macros"]
+members = [
```

Comment:
how long term should the separation exist? My current logic of having 3 separate parts were:
  - `data-model` separate because we may have different parsers for it - currently matter idl files, however in the future we may move towards a common CSA-supported XML format 
  - `idl-parser` because parsing IDLs seems to be potentially useful in generall. My current gripe with our CPP codegen is that zap takes O(10-30 seconds), python is O(0.5-5seconds) and rust can parse an entire "full all cluster IDL" in 10ms ... so I could potentially want to use this (idl + dm) outside the RS 
  - `codegen` separate because rust-specific (given that I may want IDL and DM outside). This is also completely separate from a `macros` crate to enable unit testing

Right now I am thinking of a path of:
  - data model can be one crate, with `IDL` as a feature
  - keep codegen separate, but better named - it is generally macros really. I do not like names like `util`/`support`/`helpers` as they are just meaningless fillers. Will have to figure out a better name for it.
  - macros then references data_model and codegen (whatever they get named)
  
 Does that sound right? Any other suggestions on layout and logic? I do not have much experience in breaking down larger rust projects.

Reply:
First attempt: updated into data-model and macros-impl, with the reasoning that data model may be used independently outside macros for any tooling that wants some data model access. So far I can think of rust-based codegen (instead of existing python or javascript) or any automation/scripting logic.

Reply:
OK. I think I'm fine with that actually!